packages/nimble-components/src/utilities/models/tracker.ts

```diff
@@ -0,0 +1,62 @@
+type ObjectFromList<T extends readonly string[], V = string> = {
+    [K in T extends readonly (infer U)[] ? U : never]: V;
+};
+
+/**
+ * Generic Tracker which sets or resets flags and utilities
+ */
+export abstract class Tracker<WhimsList extends readonly string[]> {
```

Comment:
A "whim" is an unfamiliar term to me (unless it's [this definition](https://www.merriam-webster.com/dictionary/whim), which I doubt).

Reply:
Yeah, I was out of ideas there and asked ChatGPT 😅, I 'm open to any suggestions

Reply:
At the moment `trackedItemList` seems more appropriate (and for the `_whims` field it could be `trackedItemsState`?). However, see my other comment below for concern around this class in general.